src/main/java/org/logstash/beats/Server.java

```diff
@@ -0,0 +1,137 @@
+        workGroup = new NioEventLoopGroup();
+
+
+    }
+
+    public void enableSSL(SslSimpleBuilder builder) {
+        sslBuilder = builder;
+    }
+
+    public Server listen() throws InterruptedException {
+        BeatsInitializer beatsInitializer = null;
+
+        try {
+            logger.info("Starting server listing port: " + this.port);
+
+            beatsInitializer = new BeatsInitializer(this);
+
+            ServerBootstrap server = new ServerBootstrap();
+            server.group(bossGroup, workGroup)
+                    .channel(NioServerSocketChannel.class)
+                    .handler(new LoggingHandler())
+                    .childHandler(beatsInitializer);
+
+            Channel channel = server.bind(port).sync().channel();
+            channel.closeFuture().sync();
+        } finally {
+            beatsInitializer.shutdownEventExecutor();
+
+            bossGroup.shutdownGracefully(0, SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            workGroup.shutdownGracefully(0, SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        }
+
+        return this;
+    }
+
+    public void stop() throws InterruptedException {
+        logger.debug("Server shutdown...");
+
+        Future<?> bossWait = bossGroup.shutdownGracefully(0, SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        Future<?> workWait = workGroup.shutdownGracefully(0, SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        bossWait.await();
+        workWait.await();
+        logger.debug("Server stopped");
+    }
+
+    public void setMessageListener(IMessageListener listener) {
+        messageListener = listener;
+    }
+
+    public boolean isSslEnable() {
+        if(this.sslBuilder != null) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private class BeatsInitializer extends ChannelInitializer<SocketChannel> {
+        private final String LOGGER_HANDLER = "logger";
+        private final String SSL_HANDLER = "ssl-handler";
+        private final String KEEP_ALIVE_HANDLER = "keep-alive-handler";
+        private final String BEATS_PARSER = "beats-parser";
+        private final String BEATS_HANDLER = "beats-handler";
+        private final int DEFAULT_IDLESTATEHANDLER_THREAD = 4;
+        private final EventExecutorGroup idleExecutorGroup;
+        private final BeatsHandler beatsHandler;
+        private final LoggingHandler loggingHandler = new LoggingHandler();
+        private final Server server;
+
+        public BeatsInitializer(Server s) {
+            server = s;
+            beatsHandler = new BeatsHandler(server.messageListener);
+            idleExecutorGroup = new DefaultEventExecutorGroup(DEFAULT_IDLESTATEHANDLER_THREAD);
+        }
+
+        public void initChannel(SocketChannel socket) throws IOException, NoSuchAlgorithmException, CertificateException {
+            ChannelPipeline pipeline = socket.pipeline();
+
+            pipeline.addLast(LOGGER_HANDLER, loggingHandler);
```

Comment:
Should only be one, left over when I've created the initializer class. good catch.